src/AnalysisPrograms/Recognizers/PteropusSpecies.cs

```diff
@@ -0,0 +1,450 @@
+                log.Warn("Could not access Territorial configuration parameters");
+            }
+
+            RecognizerResults results2 = null;
+            if (ConfigFile.TryGetProfile(genericConfig, "Wingbeats", out var profile2))
+            {
+                results2 = WingBeats(audioRecording, genericConfig, profile2, outputDirectory, segmentStartOffset);
+                log.Info("Wingbeat event count = " + results2.Events.Count);
+            }
+            else
+            {
+                log.Warn("Could not access Wingbeats configuration parameters");
+            }
+
+            // combine the results
+            if (results1 != null && results2 != null)
+            {
+                results1.Events.AddRange(results2.Events);
+                results1.Plots.AddRange(results2.Plots);
+            }
+
+            //Set following true if you want special debug spectrogram, i.e. with special plots
+            //In addition, standard spectrograms are produced when you set true in the config file, Towsey.PteropusSpecies.yml.
+            if (false)
+            {
+                //var image = sonogram.GetImageFullyAnnotated("Test");
+                string speciesName = genericConfig[AnalysisKeys.SpeciesName] ?? "Pteropus species";
+                var image = SpectrogramTools.GetSonogramPlusCharts(results1.Sonogram, results1.Events, results1.Plots, null);
+                var opPath = outputDirectory.Combine(FilenameHelpers.AnalysisResultName(Path.GetFileNameWithoutExtension(audioRecording.BaseName), speciesName, "png", "DebugSpectrogram"));
+                string imageFilename = audioRecording.BaseName + ".profile.png";
+                image.Save(Path.Combine(outputDirectory.FullName, imageFilename));
+            }
+
+            return results1;
+        }
+
+        /// <summary>
+        /// THis method does the work.
+        /// </summary>
+        /// <param name="audioRecording">the recording.</param>
+        /// <param name="configuration">the config file.</param>
+        /// <param name="outputDirectory">where results are to be put.</param>
+        /// <param name="segmentStartOffset">where one segment is located in the total recording.</param>
+        /// <returns>a list of events.</returns>
+        internal static RecognizerResults TerritorialCall(AudioRecording audioRecording, Config configuration, Config profile, DirectoryInfo outputDirectory, TimeSpan segmentStartOffset)
+        {
+            // get the common properties
+            string speciesName = configuration[AnalysisKeys.SpeciesName] ?? "Pteropus species";
+            string abbreviatedSpeciesName = configuration[AnalysisKeys.AbbreviatedSpeciesName] ?? "Pteropus";
+
+            // The following parameters worked well on a ten minute recording containing 14-16 calls.
+            // Note: if you lower the dB threshold, you need to increase maxDurationSeconds
+            int minHz = profile.GetIntOrNull(AnalysisKeys.MinHz) ?? 800;
```

Comment:
It's concerning that this is not strongly typed.

Reply:
Can you give me an example of what you would like.

Reply:
https://github.com/QutEcoacoustics/audio-analysis/blob/master/src/AnalysisPrograms/AcousticIndices.cs#L73